### ISSUE TITLE
Logistic regression

### ISSUE THREAD

**Issue by amirziai** [ap307/logistic_regression_example#1]
Is the `pretty_ad_output = np.exp(pretty_ad_output)` line correct?

**Comment by amirziai** [comment]
Take a look at:
https://www.wikiwand.com/en/Logistic_regression
https://github.com/tpeng/coursera-ml-python/blob/master/03-logistic-regression.ipynb
https://gist.github.com/vietjtnguyen/6655020

**Comment by ap307** [comment]
From the wiki reference, assumed that the probabilities should be normalized by dividing by sum of exp(beta dot x) of other features. Normalization was meant to occur in the next line of code.

**Comment by amirziai** [comment]
Since we have more than 2 classes here there are a few different ways that this can be done:
- softmax
- One vs rest
- One vs one
See the reference here:
http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.predict_proba

and the scikit-learn material here:
http://scikit-learn.org/stable/modules/multiclass.html

The default implementation is OvR. See if this helps in understanding what's going on:
https://gist.github.com/amirziai/c1ead66bc0492a729de0406d2a413651
